torch/_subclasses/fake_tensor.py

```diff
@@ -424,26 +424,31 @@ def nonzero(fake_mode, func, arg):
     ):
         # Without symints/symfloats, cannot handle this
         raise DynamicOutputShapeException(func)
-    nnz = fake_mode.shape_env.create_unbacked_symint()
 
-    from torch.fx.experimental.symbolic_shapes import (
-        constrain_range,
-        definitely_true,
-        guard_int,
-    )
+    if arg.nonzero_memo is None:
+        from torch.fx.experimental.symbolic_shapes import (
+            constrain_range,
+            definitely_true,
+            guard_int,
+        )
+
+        nnz = fake_mode.shape_env.create_unbacked_symint()
+
+        # This is unsound, but it works well in practice
+        # See https://docs.google.com/document/d/1lFRYAJo5nrfxRhwIzGnfi2pbLpU6T4ytSRSuLJ5qebI/edit#
+        # TODO: Add a config knob to turn off this unsound behavior
+        lower = 2
+        upper = None
+        # But don't give totally unsatisfiable bounds if we know it's too small!
+        if definitely_true(arg.numel() < 2):
+            lower = 0
+            upper = guard_int(arg.numel())
```

Comment:
If we're guarding on the numel, we might as well make `lower = upper`.

Reply:
guarding on numel of the INPUT. That won't tell you the lower bound